nameko/messaging.py

```diff
@@ -194,6 +199,9 @@ def publish(self, propagating_headers, msg, **kwargs):
         retry = kwargs.pop('retry', self.retry)
         retry_policy = kwargs.pop('retry_policy', self.retry_policy)
 
+        declare = self.declare[:]
+        declare.extend(kwargs.pop('declare', ()))
```

Comment:
It's never going to have any values appended to it (it's only going to exist long enough is to be iterated over by `extend`) so a tuple is marginally simpler.